wp-graphql/wp-graphql
# Chore: PHPStan 1.6 fixes (secondpass)

What does this implement/fix? Explain your changes.
---------------------------------------------------
This PR builds off of #2360 to fix another 51 errors in PHPStan 1.6.x. Unlike that PR, this one includes changes to the actual code.

To make merging easier #2360 has been squashed into this PR.

**Note**: the dependencies themselves are still not updated in this PR, to allow for merging before the remaining errors are resolved.


### Details
- stan: Remove "impossible" unnecessary conditionals (e..g `isset` in a property that _must_ exist ), and add some missing type hints.
- fix: `Model\MenuItem::init()` now correctly checks for `$this->fields` before initializing.
- fix: The specific `WP_Error` messages are now thrown during a failed `MediaItemCreate` and `MediaItemUpdate` mutation.

### To Test
- On develop, `run composer require --dev szepeviktor/phpstan-wordpress phpstan/phpstan`, followed by `composer phpstan`.
- Checkout this PR, and then run `composer require --dev szepeviktor/phpstan-wordpress phpstan/phpstan`, followed by `composer phpstan`.
- Confirm the number of PHPStan errors dropped from 136 to 20


Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
![image](https://user-images.githubusercontent.com/29322304/166679756-6f163895-9332-4037-9395-eef5b9a09268.png)



Any other comments?
-------------------



Where has this been tested?
---------------------------
**Operating System:** Ubuntu 20.04 (wsl2 + devilbox + php 8.0.15)

**WordPress Version:** 5.9.3
